src/WebJobs.Script.WebHost/Diagnostics/Extensions/ScriptHostServiceSecurityExtension.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics.Extensions
+{
+    public static class ScriptHostServiceSecurityExtension
+    {
+        // EventId range is 600-699
+
+        private static readonly Action<ILogger, string, Exception> _defaultSecretRepoError =
```

Comment:
So we log the operation in which exception was thrown. But why can't we just read call stack of the exception to understand the operation? Did you check it?